psydac/linalg/utilities.py

```diff
@@ -48,17 +48,20 @@ def array_to_stencil(x, Xh):
     return u
 
 def petsc_to_psydac(vec, Xh):
-    """ converts a petsc Vec object to a StencilVector or a BlockVector format"""
+    """ converts a petsc Vec object to a StencilVector or a BlockVector format.
+        We gather the petsc global vector in all the processes and extract the chunck owned by the Psydac Vector.
+    """
```

Comment:
Typo: chunck -> chunk

If I understand correctly this docstring, we can only use this function if the global vector fits in memory. This may not be the case if we are running a large parallel simulation. Please add an explanation and possibly a warning to the user.

Reply:
I have added warning section in the docstring [642acaf](https://github.com/pyccel/psydac/pull/244/commits/642acaf68423401ee656702996cca0000191b066), I'm not sure we want to raise a warning every time this function is used.

Reply:
Sounds good to me